scheduler/feasible.go

```diff
@@ -192,14 +204,29 @@ func (iter *ProposedAllocConstraintIterator) Next() *structs.Node {
 		// Get the next option from the source
 		option := iter.source.Next()
 
-		// Hot-path if the option is nil or there are no distinct_hosts constraints.
-		if option == nil || !(iter.jobDistinctHosts || iter.tgDistinctHosts) {
+		// Hot-path if the option is nil or there are no distinct_hosts or
+		// distinct_property constraints.
+		hosts := iter.jobDistinctHosts || iter.tgDistinctHosts
+		properties := iter.distinctProperties.HasDistinctPropertyConstraints()
```

Comment:
Why not call `HasDistinctPropertyConstraints()` in `SetJob` and then just check for a `nil` `distinctProperties` here? Seems more efficient that way, avoids some allocations.